src/main/java/de/caritas/cob/agencyservice/api/service/TenantHibernateInterceptor.java

```diff
@@ -13,16 +15,52 @@
 @Slf4j
 public class TenantHibernateInterceptor extends EmptyInterceptor {
 
+  private static Boolean multiTenancyEnabled;
+
   @Override
   public void preFlush(Iterator entities) {
-    Object entity;
-    while (entities.hasNext()) {
-      entity = entities.next();
-      if (entity instanceof TenantAware) {
-        ((TenantAware) entity).setTenantId(TenantContext.getCurrentTenant());
+
+    if (multiTenancyEnabled == null) {
+      multiTenancyEnabled = resolveMultiTenancy();
+    }
+
+    if (multiTenancyEnabled) {
+      Object entity;
+      while (entities.hasNext()) {
+        entity = entities.next();
+        if (entity instanceof TenantAware) {
+          ((TenantAware) entity).setTenantId(TenantContext.getCurrentTenant());
+        }
       }
     }
 
     super.preFlush(entities);
   }
+
+  private boolean resolveMultiTenancy() {
+    var multiTenancyValue = getParamFromEnvVariable();
+
+    if (multiTenancyValue == null) {
+      multiTenancyValue = getParamFromPropertyFile();
+    }
+
+    return Boolean.parseBoolean(multiTenancyValue);
+  }
+
+  private String getParamFromPropertyFile() {
+    try {
+      Properties prop = new Properties();
+      prop.load(TenantHibernateInterceptor.class.getClassLoader()
+          .getResourceAsStream("application-local.properties"));
+      return prop.getProperty("multitenancy.enabled");
+
+    } catch (IOException ex) {
+      return null;
```

Comment:
please throw RuntimeException in this case

Reply:
done